corehq/apps/case_search/xpath_functions/__init__.py

```diff
@@ -0,0 +1,15 @@
+from .query_functions import not_
+from .subcase_functions import subcase
+from .value_functions import date  # noqa: F401
+
+# functions that transform a value
+XPATH_VALUE_FUNCTIONS = {
+    'date': date,
+}
+
+
+XPATH_QUERY_FUNCTIONS = {
+    'not': not_,
+    'subcase-exists': subcase,
+    'subcase-count': subcase,
+}
```

Comment:
I'll need to spend a bit more time with this PR to wrap my head around it, but could you explain a little about the reasons for splitting apart these two types of functions?

Reply:
This is to distinguish between functions that return an ES filter and functions that mutate value:
```
# date is the only value function we support currently
dob > date('2022-01-01')

# but you can't use it on it's own, it has to be as part of another expression
date('2022-01-01')

# but you can use functions like not
not(a = 1)
```